go/vt/vtgate/vstream_manager.go

```diff
@@ -676,6 +696,24 @@ func (vs *vstream) sendAll(ctx context.Context, sgtid *binlogdatapb.ShardGtid, e
 	return nil
 }
 
+// generateFullyCopyCompleted returns a new copy_completed event only if all stream has received a copy_completed event.
+// This new event represents the completion of all the copy operations.
+func (vs *vstream) generateFullyCopyCompleted(ctx context.Context, sgtid *binlogdatapb.ShardGtid, event *binlogdatapb.VEvent) *binlogdatapb.VEvent {
```

Comment:
IMO it would be a little more clear if it was ~:
```
func (vs *vstream) isCopyFullyCompleted(ctx context.Context, sgtid *binlogdatapb.ShardGtid, event *binlogdatapb.VEvent) (bool, *binlogdatapb.VEvent) {
```

Then you would ignore the event if we're not done:
```
if fullyCopied, doneEvent := vs.isCopyFullyCompleted(ctx, sgtid, event); fullyCopied {
```

Current code is fine too though.

Reply:
Your suggestion makes sense. I fixed it by f4ac20ab297a91f65ce28eab065f4fd468d48875.